sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,247 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = "sale.order.import"
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    # Make doc_type field support for EDIFACT type
+    doc_type = fields.Selection(readonly=False)
+    parsed_json = fields.Text()
+    edifact_ok = fields.Boolean(compute="_compute_edifact_ok", store=True)
+    # Techincal fields used to get the extension part of files
+    file_ext = fields.Char(default="txt,d96a")
+    # EDIFACT format release
+    release = fields.Char(default="d96a")
+
+    @api.depends("order_file", "order_filename")
+    def _compute_edifact_ok(self):
+        for rec in self:
+            if not (rec.order_file and rec.order_filename):
+                rec.edifact_ok = False
+                continue
+            extensions = rec.file_ext.split(",")
+            path, ext = os.path.splitext(self.order_filename)
+            ok = ext and ext[1:] in extensions
+            if not ok:
+                ok = b64decode(self.order_file[:4]) == b"UNB"
+            rec.edifact_ok = ok
+
+    def button_parse_order_preview(self):
+        """
+        This method is called by the button to take a quick look at the pydifact
+        structure ('edifact' type) and the parsed object ('order') from
+        the original document before importing the document.
+        """
```

Comment:
```suggestion
        """
        Generate a preview of the data before importing.

        This method is called by the button to take a quick look at the pydifact
        structure ('edifact' type) and the parsed object ('order') from
        the original document before importing the document.
        """
```
https://peps.python.org/pep-0257/#multi-line-docstrings